lLyrics/LocalParser.py

```diff
@@ -0,0 +1,105 @@
+# Parses lyrics from a .lyric or .lrc file or from the tags of the song
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import urllib
+import os.path as path
+from mutagen.oggvorbis import OggVorbis
+from mutagen.oggopus import OggOpus
+from mutagen.flac import FLAC
+from mutagen.id3 import ID3
+
+
+class Parser(object):
+    def __init__(self, artist, title, song_path):
+        self.artist = artist
+        self.title = title
+        self.lyrics = ""
+        self.song_path = song_path
+
+    def parse(self):
+        # convert the song url to a usable path
+        path = self.song_path
+        path = urllib.parse.unquote(path)
+        path = path.replace("file://", "")
+        dir = path[0:((path.rfind("/")) + 1)]
+        file_name = path[((path.rfind("/")) + 1):path.rfind(".")]
+        out = ""
+        if path.endswith("mp3"):
+            out = self.get_id3_lyrics(path)
+        elif path.endswith("ogg"):
+            out = self.get_ogg_lyrics(path)
+        elif path.endswith("opus"):
+            out = self.get_opus_lyrics(path)
+        elif path.endswith("flac"):
+            out = self.get_flac_lyrics(path)
+        if out is None:
+            file = self.check_for_file(dir, file_name)
```

Comment:
Name suggests it only checks, but in fact it opens a path and returns opened file.  Since `open` calls are not guarded anyway, wouldn't it make more sense to return a string path here, and call `open` only once?

Reply:
yeah returning open() is kinda sketchy anyway.